Build fails on `Failed to start Grizzly HTTP server: Address already in use`

I'm trying to do a `mvn install` however this fails with an exception of:

**Failed to start Grizzly HTTP server: Address already in use**:

```
 1> 11:47:54,398 INFO  [f.p.e.c.f.c.v.ElasticsearchClientV7] Elasticsearch Client for version 7.x connected to a node running version 7.1.0
  2> REPRODUCE WITH:
  2> mvn integration-test -Dtests.seed=E0E66723398841EC -Dtests.class=fr.pilato.elasticsearch.crawler.fs.test.integration.FsCrawlerRestIT -Dtests.method="testDocumentWithExternalTags" -Dtests.locale=lv-LV -Dtests.timezone=Africa/Gaborone
ERROR   0.84s J2 | FsCrawlerRestIT.testDocumentWithExternalTags <<<
   > Throwable #1: javax.ws.rs.ProcessingException: Failed to start Grizzly HTTP server: Address already in use
   > 	at __randomizedtesting.SeedInfo.seed([E0E66723398841EC:4F75B14AC5992333]:0)
   > 	at org.glassfish.jersey.grizzly2.httpserver.GrizzlyHttpServerFactory.createHttpServer(GrizzlyHttpServerFactory.java:270)
   > 	at org.glassfish.jersey.grizzly2.httpserver.GrizzlyHttpServerFactory.createHttpServer(GrizzlyHttpServerFactory.java:93)
   > 	at fr.pilato.elasticsearch.crawler.fs.rest.RestServer.start(RestServer.java:61)
   > 	at fr.pilato.elasticsearch.crawler.fs.test.integration.AbstractRestITCase.startRestServer(AbstractRestITCase.java:47)
   > 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
   > 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
   > 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
   > 	at java.lang.reflect.Method.invoke(Method.java:498)
   > 	at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1737)
   > 	at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:968)
   > 	at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:984)
   > 	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:55)
   > 	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
   > 	at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)
   > 	at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:817)
   > 	at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:468)
   > 	at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:943)
   > 	at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:829)
   > 	at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:879)
   > 	at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:890)
   > 	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:48)
   > 	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
   > 	at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)
   > 	at java.lang.Thread.run(Thread.java:748)
   > Caused by: java.net.BindException: Address already in use
   > 	at sun.nio.ch.Net.bind0(Native Method)
   > 	at sun.nio.ch.Net.bind(Net.java:433)
   > 	at sun.nio.ch.Net.bind(Net.java:425)
   > 	at sun.nio.ch.ServerSocketChannelImpl.bind(ServerSocketChannelImpl.java:223)
   > 	at sun.nio.ch.ServerSocketAdaptor.bind(ServerSocketAdaptor.java:74)
   > 	at org.glassfish.grizzly.nio.transport.TCPNIOBindingHandler.bindToChannelAndAddress(TCPNIOBindingHandler.java:107)
   > 	at org.glassfish.grizzly.nio.transport.TCPNIOBindingHandler.bind(TCPNIOBindingHandler.java:64)
   > 	at org.glassfish.grizzly.nio.transport.TCPNIOTransport.bind(TCPNIOTransport.java:215)
   > 	at org.glassfish.grizzly.nio.transport.TCPNIOTransport.bind(TCPNIOTransport.java:195)
   > 	at org.glassfish.grizzly.nio.transport.TCPNIOTransport.bind(TCPNIOTransport.java:186)
   > 	at org.glassfish.grizzly.http.server.NetworkListener.start(NetworkListener.java:711)
   > 	at org.glassfish.grizzly.http.server.HttpServer.start(HttpServer.java:256)
   > 	at org.glassfish.jersey.grizzly2.httpserver.GrizzlyHttpServerFactory.createHttpServer(GrizzlyHttpServerFactory.java:267)
   > 	... 23 more
  1> 11:47:55,217 INFO  [f.p.e.c.f.t.f.AbstractFSCrawlerTestCase]   --> Launching test [test_all_documents_with_rest]
  1> 11:47:55,238 INFO  [f.p.e.c.f.c.v.ElasticsearchClientV7] Elasticsearch Client for version 7.x connected to a node running version 7.1.0
  2> REPRODUCE WITH:
  2> mvn integration-test -Dtests.seed=E0E66723398841EC -Dtests.class=fr.pilato.elasticsearch.crawler.fs.test.integration.FsCrawlerRestIT -Dtests.method="testAllDocumentsWithRest" -Dtests.locale=lv-LV -Dtests.timezone=Africa/Gaborone
```

I'm not sure which port it is trying to listen on though?

--- a/integration-tests/it-common/src/main/java/fr/pilato/elasticsearch/crawler/fs/test/integration/AbstractITCase.java
+++ b/integration-tests/it-common/src/main/java/fr/pilato/elasticsearch/crawler/fs/test/integration/AbstractITCase.java
@@ -238,7 +238,7 @@ public abstract class AbstractITCase extends AbstractFSCrawlerTestCase {
     @BeforeClass
     public static void startElasticsearchRestClient() throws IOException {
         String testClusterCloudId = System.getProperty("tests.cluster.cloud_id");
-        if (testClusterCloudId != null) {
+        if (testClusterCloudId != null && !testClusterCloudId.isEmpty()) {
             testClusterUrl = decodeCloudId(testClusterCloudId);
             staticLogger.debug("Using cloud id [{}] meaning actually [{}]", testClusterCloudId, testClusterUrl);
         } else {

--- a/integration-tests/it-common/src/main/java/fr/pilato/elasticsearch/crawler/fs/test/integration/AbstractITCase.java
+++ b/integration-tests/it-common/src/main/java/fr/pilato/elasticsearch/crawler/fs/test/integration/AbstractITCase.java
@@ -243,6 +243,10 @@ public abstract class AbstractITCase extends AbstractFSCrawlerTestCase {
             staticLogger.debug("Using cloud id [{}] meaning actually [{}]", testClusterCloudId, testClusterUrl);
         } else {
             testClusterUrl = System.getProperty("tests.cluster.url", DEFAULT_TEST_CLUSTER_URL);
+            if (testClusterUrl.isEmpty()) {
+                // When running from Maven CLI, tests.cluster.url is empty and not null...
+                testClusterUrl = DEFAULT_TEST_CLUSTER_URL;
+            }
         }
 
         staticLogger.info("Starting a client against [{}]", testClusterUrl);

--- a/integration-tests/it-v6/pom.xml
+++ b/integration-tests/it-v6/pom.xml
@@ -180,7 +180,7 @@
                                             <discovery.type>single-node</discovery.type>
                                             <xpack.license.self_generated.type>trial</xpack.license.self_generated.type>
                                             <xpack.security.enabled>true</xpack.security.enabled>
-                                            <ELASTIC_PASSWORD>${integ.security.password}</ELASTIC_PASSWORD>
+                                            <ELASTIC_PASSWORD>${tests.cluster.pass}</ELASTIC_PASSWORD>
                                         </env>
                                     </build>
                                     <run>

--- a/integration-tests/it-v7/pom.xml
+++ b/integration-tests/it-v7/pom.xml
@@ -180,7 +180,7 @@
                                             <discovery.type>single-node</discovery.type>
                                             <xpack.license.self_generated.type>trial</xpack.license.self_generated.type>
                                             <xpack.security.enabled>true</xpack.security.enabled>
-                                            <ELASTIC_PASSWORD>${integ.security.password}</ELASTIC_PASSWORD>
+                                            <ELASTIC_PASSWORD>${tests.cluster.pass}</ELASTIC_PASSWORD>
                                         </env>
                                     </build>
                                     <run>

--- a/pom.xml
+++ b/pom.xml
@@ -68,9 +68,12 @@
         <!-- For integration tests using Docker or external cluster -->
         <integ.elasticsearch.image>docker.elastic.co/elasticsearch/elasticsearch</integ.elasticsearch.image>
         <integ.elasticsearch.version>${elasticsearch.version}</integ.elasticsearch.version>
-        <integ.security.username>elastic</integ.security.username>
-        <integ.security.password>changeme</integ.security.password>
         <integ.elasticsearch.port>9200</integ.elasticsearch.port>
+        <tests.cluster.url></tests.cluster.url>
+        <tests.cluster.cloud_id></tests.cluster.cloud_id>
+        <tests.cluster.user>elastic</tests.cluster.user>
+        <tests.cluster.pass>changeme</tests.cluster.pass>
+        <tests.rest.port>8080</tests.rest.port>
 
         <!-- Randomized testing framework -->
         <tests.locale>random</tests.locale>
@@ -192,6 +195,11 @@
                             <tests.timezone>${tests.timezone}</tests.timezone>
                             <java.util.logging.manager>org.apache.logging.log4j.jul.LogManager</java.util.logging.manager>
                             <java.awt.headless>true</java.awt.headless>
+                            <tests.cluster.url>${tests.cluster.url}</tests.cluster.url>
+                            <tests.cluster.cloud_id>${tests.cluster.cloud_id}</tests.cluster.cloud_id>
+                            <tests.cluster.user>${tests.cluster.user}</tests.cluster.user>
+                            <tests.cluster.pass>${tests.cluster.pass}</tests.cluster.pass>
+                            <tests.rest.port>${tests.rest.port}</tests.rest.port>
                         </systemProperties>
                     </configuration>
 

